src/sqltest/platform/dialog/dialogPane.test.ts

```diff
@@ -4,124 +4,101 @@
 	});
 
 	test('Creating a pane from content without tabs initializes the model view content correctly', () => {
 		// If I fill in a dialog's content with the ID of a specific model view provider and then render the dialog
 		let modelViewId = 'test_content';
+		let bootstrapCalls = 0;
+		setupBootstrap((collection, moduleType, container, selectorString, params: DialogComponentParams, input, callbackSetModule) => {
+			assert.equal(params.modelViewId, modelViewId);
+			bootstrapCalls++;
+		});
 		dialog.content = modelViewId;
-		let dialogPane = new DialogPane(dialog.title, dialog.content, () => undefined, mockInstantiationService.object);
+		let dialogPane = new DialogPane(dialog.title, dialog.content, () => undefined, undefined);
 		dialogPane.createBody(container);
-
-		// Then a single dialog-modelview-container element is added directly to the dialog pane
-		mockInstantiationService.verify(x => x.invokeFunction(
-			It.isAny(),
-			It.isAny(),
-			It.isAny(),
-			It.isAny(),
-			It.is((x: DialogComponentParams) => x.modelViewId === modelViewId),
-			undefined,
-			It.isAny()), Times.once());
+		assert.equal(bootstrapCalls, 1);
 	});
 
 	test('Creating a pane from content with a single tab initializes without showing tabs', () => {
 		// If I fill in a dialog's content with a single tab and then render the dialog
 		let modelViewId = 'test_content';
+		let bootstrapCalls = 0;
+		setupBootstrap((collection, moduleType, container, selectorString, params: DialogComponentParams, input, callbackSetModule) => {
+			assert.equal(params.modelViewId, modelViewId);
+			bootstrapCalls++;
+		});
 		dialog.content = [new DialogTab('', modelViewId)];
-		let dialogPane = new DialogPane(dialog.title, dialog.content, () => undefined, mockInstantiationService.object);
-		dialogPane.createBody(container);
-
-		// Then a single dialog-modelview-container element is added directly to the dialog pane
-		mockInstantiationService.verify(x => x.invokeFunction(
-			It.isAny(),
-			It.isAny(),
-			It.isAny(),
-			It.isAny(),
-			It.is((x: DialogComponentParams) => x.modelViewId === modelViewId),
-			undefined,
-			It.isAny()), Times.once());
-	});
-
-	test('Creating a pane from content with multiple tabs initializes multiple model view tabs', () => {
-		// If I fill in a dialog's content with a single tab and then render the dialog
-		let modelViewId1 = 'test_content_1';
-		let modelViewId2 = 'test_content_2';
-		dialog.content = [new DialogTab('tab1', modelViewId1), new DialogTab('tab2', modelViewId2)];
-		let dialogPane = new DialogPane(dialog.title, dialog.content, () => undefined, mockInstantiationService.object);
+		let dialogPane = new DialogPane(dialog.title, dialog.content, () => undefined, undefined);
 		dialogPane.createBody(container);
-
-		// Then a dialog-modelview-container element is added for the first tab (subsequent ones get added when the tab is actually clicked)
-		mockInstantiationService.verify(x => x.invokeFunction(
```

Comment:
@MattIrv I had to remove this test. It is fundamentally broken. In master this function is being called multiple times even though the test states it should only be called once. I'm not sure why typemoq was not failing this test. But if you step through the logic of createBody it makes sense that it is being called for every tab on creation, rather than when the tab is clicked as the comment states.

Reply:
You're right that it gets called for each tab. The comment was out of date and I missed updating it when I made the change to initialize all tabs immediately.

I think it was passing because of the match on `x.modelViewId === modelViewId1`. It's true that the function gets called twice, but only once with that particular model view ID (though I probably should have updated the test to check that it gets called for both model view IDs when I made the change to make it be called twice)